janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/ModifiableConfiguration.java

```diff
@@ -34,7 +34,8 @@ public ModifiableConfiguration(ConfigNamespace root, WriteConfiguration config,
 
     public<O> ModifiableConfiguration set(ConfigOption<O> option, O value, String... umbrellaElements) {
         verifyOption(option);
-        Preconditions.checkArgument(!option.isFixed() || !isFrozen(), "Cannot change configuration option: %s", option);
+        Preconditions.checkArgument(option.getName() == "storage-version" ||
```

Comment:
1. I would use a constant for `"storage-version"` (something like `private final static String STORAGE_VERSION = "storage-version";`). 
2. Are you sure that `option.getName()` is interned? I would use `STORAGE_VERSION.equals(option.getName())` if we don't need much speed in this method. If we do need to increase speed in this method I would change `ConfigElement` constructor line `this.name = name;` to line `this.name = name.intern();` (but it has to be reviewed. I'm not sure.)

Reply:
+1 on `STORAGE_VERSION.equals(option.getName())` for the string comparison, but it's not good to have the upgradeable options hardcoded in this class. It would be better to pass in a list of upgradeable options on a constructor.